combat/CombatTargetting.cpp

```diff
@@ -0,0 +1,150 @@
+#include "CombatTargetting.h"
+
+#include "../universe/Condition.h"
+#include "../universe/Enums.h"
+#include "../universe/Planet.h"
+#include "../universe/Fleet.h"
+#include "../universe/Ship.h"
+#include "../universe/ShipDesign.h"
+#include "../universe/Fighter.h"
+#include "../universe/ValueRef.h"
+
+#include "../util/Logger.h"
+#include "../util/Random.h"
+
+bool Targetting::isPreferredTarget(Targetting::TriggerConditions conditions,
+                                   Targetting::Target target) {
+    const Precision highest_weight = Targetting::findPrecision(conditions);
+    for (long unsigned int i = 0; i < conditions.conditions.size(); i++) {
+        const auto& condition = conditions.conditions[i];
+        if (!condition) {
+            DebugLogger() << "No preferences. Target is perfect." ;
+            return true;
+        }
+
+        if (Targetting::isPreferredTarget(condition, target)) {
+            const int upper = std::round(highest_weight * 1.5f); // HEURISTIC
+            const int improbability = RandInt(0, upper);
+            const int luck = conditions.weights[i];
+            DebugLogger() << "Prioritized locking attempt" <<  [&improbability, &luck, &upper](){
+                                                                   std::stringstream ss;
+                                                                   ss << "( " << luck << "/" << upper << " ?>? " << improbability << "/" << upper << ")";
+                                                                   return ss.str();
+                                                               }();
+            if (luck > improbability) {
+                DebugLogger() << "Locked in target.";
+                return true;
+            } else {
+                DebugLogger() << "Couldnt acquire lock." ;
+            }
+        }
+    }
+    return false;
+}
+
+bool Targetting::isPreferredTarget(Targetting::TriggerCondition condition,
+                                   Targetting::Target target) {
+    if (!condition) {
+            DebugLogger() << "No preferences. Target is perfect." ;
+            return true;
+    }
+
+    DebugLogger() << "Evaluate preferred prey condition" << condition->Description();
+    bool is_preferred = condition->Eval(target);
+    auto obj_type = target->ObjectType();
+    if ( is_preferred ) {
+        DebugLogger() << "This " << obj_type << " ship should die! " ;
+    } else {
+        DebugLogger() << "Don't care about this " << obj_type << " - reroll if possible. " ;
+    }
+    return is_preferred;
+}
+Targetting::TriggerCondition Targetting::preyAsTriggerCondition(const Targetting::Prey prey) {
+        switch( prey ) {
```

Comment:
be consistent with `switch(...)` or `switch (...)`. I forget which is preferred based on existing code.

`( prey )` -> `(prey)`